docs/self-managed/setup/deploy/amazon/amazon-eks/irsa.md

```diff
@@ -311,31 +453,100 @@ module "opensearch_role" {
+
+Annotate the service account with the `iam_role_arn` output.
+
+  </TabItem>
 
-The output of the module `opensearch_role` has the output `iam_role_arn` to annotate a service account to use the mapping.
+  <TabItem value="aws-cli" label="AWS CLI">
 
-Annotate the service account with the `iam_role_arn` output of the `opensearch_role`.
+To assign the policy to a role using the AWS CLI, follow these steps:
 
+1. **Create the IAM Role**:
+
+```bash
+aws iam create-role \
+  --role-name opensearch-role \
+  --assume-role-policy-document '{
+    "Version": "2012-10-17",
+    "Statement": [
+      {
+        "Effect": "Allow",
+        "Principal": {
+          "Federated": "arn:aws:iam::<ACCOUNT-ID>:oidc-provider/oidc.eks.<REGION>.amazonaws.com/id/<EKS-ID>"
+        },
+        "Action": "sts:AssumeRoleWithWebIdentity",
+        "Condition": {
+          "StringEquals": {
+            "oidc.eks.<REGION>.amazonaws.com/id/<EKS-ID>:sub": "system:serviceaccount:<NAMESPACE>:<SERVICE-ACCOUNT>"
+          }
+        }
+      }
+    ]
+  }'
+```
+
+2. **Attach the Policy to the Role**:
+
+```bash
+aws iam attach-role-policy \
+  --role-name opensearch-role \
+  --policy-arn arn:aws:iam::<ACCOUNT-ID>:policy/opensearch_policy
 ```
+
+  </TabItem>
+</Tabs>
+
+You need to annotate the service account with the `iam_role_arn`. Here is an example YAML configuration:
+
+```yaml
 apiVersion: v1
 kind: ServiceAccount
 metadata:
   annotations:
-    eks.amazonaws.com/role-arn: arn:aws:iam::account-id:role/role-name
-  name: opensearch-serviceaccount
-  namespace: opensearch-namespace
+    eks.amazonaws.com/role-arn: arn:aws:iam::<ACCOUNT-ID>:role/opensearch-role
+  name: <SERVICE-ACCOUNT>
+  namespace: <NAMESPACE>
 ```
 
+Replace `<ACCOUNT-ID>`, `<REGION>`, `<NAMESPACE>`, and `<SERVICE-ACCOUNT>` with the appropriate values for your Amazon OpenSearch Service and EKS setup.
+
 This step is required to be repeated for Tasklist and Zeebe, to grant their service accounts access to OpenSearch.
 
+#### Authorize the Mapping in OpenSearch
+
+To authorize the IAM role in OpenSearch for access, follow these steps:
+
+Use the following `curl` command to update the OpenSearch internal database and authorize the IAM role for access. Replace placeholders with your specific values:
+
+```bash
+curl -sS -u "<ES_DOMAIN_USER>:<ES_DOMAIN_PASSWORD>" \
+    -X PATCH \
+    "https://<ES_ENDPOINT>/_opendistro/_security/api/rolesmapping/all_access?pretty" \
+    -H 'Content-Type: application/json' \
+    -d'
+[
+  {
+    "op": "add",
+    "path": "/backend_roles",
+    "value": ["<ROLE-NAME>"]
+  }
+]
+'
+```
+
+- Replace `<ES_DOMAIN_USER>` and `<ES_DOMAIN_PASSWORD>` with your OpenSearch domain admin credentials.
+- Replace `<ES_ENDPOINT>` with your OpenSearch endpoint URL.
+- Replace `<ROLE-NAME>` with the IAM role name created by Terraform, which is output by the `opensearch_role` module.
```

Comment:
I'd suggest moving this point under `Via the REST API` as this is the way to do it with the REST API in case a user is using `fine-grained access control` and master/pw.

At the same time, I'd suggest to limit it to being an example and not being a best practise as you're using `username:password`. Additionally, the endpoint isn't exposed by default.

That's why the topic was left open with suggestions to read further into the topic as it highly depends on how a customer approaches it.
1. Terraform Module
2. Dashboard
3. REST API

I'd also suggest to not use the working `ES` but `OS`. While the service is called ES, we're still talking about the OS but just a side note.